apps/src/aichat/views/modelCustomization/SetupCustomization.tsx

```diff
@@ -127,6 +128,34 @@ const SetupCustomization: React.FunctionComponent = () => {
     );
   };
 
+  const sliderProps = {
+    name: 'temperature-slider',
+    value: aiCustomizations.temperature,
+    minValue: MIN_TEMPERATURE,
+    maxValue: MAX_TEMPERATURE,
+    step: SET_TEMPERATURE_STEP,
+    hideValue: true,
+    disabled: isDisabled(temperature) || readOnlyWorkspace,
+    onChange: event => {
+      const value = parseFloat(event.target.value).toFixed(1); // Round to nearest tenth.
```

Comment:
Does this not happen automatically with `step`?

Reply:
Unfortunately no. Adding/subtracting by `step` is handled here: https://github.com/code-dot-org/code-dot-org/blob/2d81e046f8b9dfea9fab172791e7b67880c079bd/apps/src/componentLibrary/slider/Slider.tsx#L196
However, when adding/subtracting `0.1` multiple times, because how decimal numbers are represented in binary, we end up with floats that are not quite correct.  cc: @levadadenys 

See screencast.

https://github.com/user-attachments/assets/5ec4a0ca-a31f-4bc8-89af-86d32616cce2

Reply:
I had a [similar](https://github.com/code-dot-org/code-dot-org/pull/61308) issue with floats in the AI drums UI's temperature slider, especially after adding +/- buttons, so ended up switching to integers.  

Perhaps the slider could do similar here, and divide only when consuming the value, making it a fixed point rather than floating point representation.

Reply:
Oh interesting. Yes - I like that bc it's more predictable. Will update.

Reply:
For the `Slider`, we use integers (range 1-10), but for the `aiCustomizations.temperature` `value`, we divide the slider value by 10.

Reply:
@breville - are you planning to use the DSCO `Slider` in music lab also?

Reply:
Agree that it will be easier if `DSCO Slider` will support this out of the box. I've created a [jira ticket](https://codedotorg.atlassian.net/browse/DESIGN2-171) to ivestigate and fix it in scope of `DSCO Slider`.  Will take a look into it once have the time!

Reply:
> @breville - are you planning to use the DSCO `Slider` in music lab also?

I'm not sure.  We have already customized it with the +/- buttons and there might be more to come.

Reply:
The DSCO slider has +/- buttons also. But if you are adding more, may be easier with customized slider.